core/src/main/java/org/radargun/stats/Message.java

```diff
@@ -0,0 +1,49 @@
+package org.radargun.stats;
+
+import java.util.concurrent.TimeUnit;
+
+import org.radargun.Operation;
+
+/**
+ * Tracks time for operations with different origin and destination nodes.
+ * The time can be tracked only using wall-clock time.
+ *
+ * @author Radim Vansa &lt;rvansa@redhat.com&gt;
+ */
+public final class Message {
+   private final Statistics statistics;
+   private long sendStartTime = Long.MAX_VALUE;
+   // private long sendCompleteTime; // hard to track when the message itself carries the timestamps
+   // private long receiveStartTime;
```

Comment:
Yes, those are expected to be present when we'll have traits with asynchronous API.